marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/IOEndpointImpl.java

```diff
@@ -274,5 +272,19 @@ public void interrupt() {
             if (this.phase == WorkPhase.RUNNING)
                 setPhase(WorkPhase.INTERRUPTING);
         }
+
+        void processOutput(InputStream[] output, Consumer<InputStream> outputConsumer) {
+            for (int i=allowsEndpointState()?1:0; i < output.length; i++) {
+                outputConsumer.accept(output[i]);
+            }
+        }
+
+        InputStream[] getOutput(InputStream[] output, Consumer<InputStream> outputConsumer) {
+            processOutput(output, outputConsumer);
```

Comment:
Would it make sense to support either asynchronous consumption of the output via an output consumer or synchronous consumption of the output via next() but not both?

One concern is avoiding the need to buffer the data from each InputStream so that we can resuse the same data for both the synchronous and asynchronous consumption. 

In other words, would it make sense to delete the call on line 283?